products/views.py

```diff
@@ -38,4 +38,28 @@ def get(self, request):
         except KeyError:
             return JsonResponse({"message": 'KeyError'}, status = 400)  
         except Product.DoesNotExist:
-            return JsonResponse({"message": 'No Data'}, status = 400)
\ No newline at end of file
+            return JsonResponse({"message": 'No Data'}, status = 400)
+
+
+class ProductDetailView(View):
+    def get(self, request, id):
+        try:
+            products = Product.objects.filter(id = id)
+            product_detail = [{
+                "id"         : product.id,
+                "name"       : product.name,
+                "description": product.description,
+                "serving"    : product.serving,
+                "cookTime"   : product.cook_time,
+                "prepTime"   : product.prep_time,
+                "price"      : int(product.price),
+                "spice"      : [spice.level for spice in product.spice_set.all()],
+                "image"      : [image.image for image in product.productimage_set.all()]
+            } for product in products]
```

Comment:
product_detail을 리스트 컴프리헨션으로 작성하신 이유가 있으신가요-?
바로 윗줄에서 오브젝트를 id(pk)값으로 필터를 한다면, 상품 디테일 API의 역할 그대로 하나의 오브젝트 만을 가져오기 때문에 굳이 리스트 컴프리헨션으로 작성하지 않으셔도 될 것 같습니다.
```suggestion
            product_detail = {
                "id"         : product.id,
                "name"       : product.name,
                "description": product.description,
                "serving"    : product.serving,
                "cookTime"   : product.cook_time,
                "prepTime"   : product.prep_time,
                "price"      : int(product.price),
                "spice"      : [spice.level for spice in product.spice_set.all()],
                "image"      : [image.image for image in product.productimage_set.all()]
            } 
```

Reply:
네 바로 수정했습니다 그리고 postman으로 해보니 오류없이 통과되었습니다 어차피 하나의값만 받으니 당연히 get 메소드로 사용해야되는데 놓첬습니다 감사합니다